src/vs/gitpod/browser/workbench/contrib/exportLogs.contribution.ts

```diff
@@ -0,0 +1,158 @@
+ *  Copyright (c) Gitpod. All rights reserved.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IBrowserWorkbenchEnvironmentService } from 'vs/workbench/services/environment/browser/environmentService';
+import { IRemoteAgentService } from 'vs/workbench/services/remote/common/remoteAgentService';
+import { IFileService } from 'vs/platform/files/common/files';
+import { Action2, MenuId, registerAction2 } from 'vs/platform/actions/common/actions';
+import { CATEGORIES } from 'vs/workbench/common/actions';
+import { ServicesAccessor } from 'vs/platform/instantiation/common/instantiation';
+import { URI } from 'vs/base/common/uri';
+import * as resources from 'vs/base/common/resources';
+import { IProgressService, ProgressLocation } from 'vs/platform/progress/common/progress';
+import { CancellationTokenSource } from 'vs/base/common/cancellation';
+import type * as zipModule from '@zip.js/zip.js';
+
+const getZipModule = (function () {
+	let zip: typeof zipModule;
+	return async () => {
+		if (!zip) {
+			// when actually importing the module change `zip.js` to `zipjs`
+			// without the dot because loader.js will do a check for `.js` extension
+			// and it won't resolve the module path correctly
+			// @ts-ignore
+			zip = await import('@zip.js/zipjs');
+			zip.configure({
+				useWebWorkers: false
+			});
+		}
+		return zip;
+	};
+})();
+
+registerAction2(class ExportLogsAction extends Action2 {
+	constructor() {
+		super({
+			id: 'gitpod.workbench.exportLogs',
+			title: { original: 'Export all logs', value: 'Export all logs' },
+			category: CATEGORIES.Developer,
+			menu: {
+				id: MenuId.CommandPalette
+			}
+		});
+	}
+
+	async run(accessor: ServicesAccessor): Promise<void> {
+		const progressService = accessor.get(IProgressService);
+		const fileService = accessor.get(IFileService);
+		const environmentService = accessor.get(IBrowserWorkbenchEnvironmentService);
+		const remoteAgentService = accessor.get(IRemoteAgentService);
+
+		const cts = new CancellationTokenSource();
+		const blobData = await progressService.withProgress<Blob | undefined>(
+			{
+				location: ProgressLocation.Dialog,
+				title: 'Exporting logs to zip file ...',
+				cancellable: true,
+				delay: 1000
+			},
+			async progress => {
+				const token = cts.token;
+
+				const zip = await getZipModule();
+				const blobWriter = new zip.BlobWriter('application/zip');
+				const writer = new zip.ZipWriter(blobWriter);
+
+				if (token.isCancellationRequested) {
+					return undefined;
+				}
+
+				const entries: { name: string; resource: URI }[] = [];
+				const logsPath = URI.file(environmentService.logsPath).with({ scheme: environmentService.logFile.scheme });
+				const stat = await fileService.resolve(logsPath);
+				if (stat.children) {
+					entries.push(...stat.children.filter(stat => !stat.isDirectory).map(stat => ({ name: resources.basename(stat.resource), resource: stat.resource })));
```

Comment:
Why sometimes we filter `isDirection` sometimes not?

Reply:
Just to filter the files that we will zip, the only case we want directories is in line 108 `const ouputLoggingDirs = stat.children.filter(stat => stat.isDirectory);` as there could be more than one inside `remoteExtHostLogsPath`